community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/RWLock.java

```diff
@@ -86,13 +86,28 @@
         // access to these is guarded by synchronized blocks
         private int readCount;
         private int writeCount;
-        private boolean movedOn;
+        private int requests = 0;
```

Comment:
what does requests represent? This is even more important to explain given the isNoRequests() method below